packages/webapp/src/containers/Filter/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+// In tempStateReducer > filterReducer
+export interface ReduxFilterEntity {
+  [filterKey: /* e.g. STATUS, LOCATION */ string]: FilterState;
+}
+
+export interface FilterState {
+  [filterOptionId: string]: /* e.g. ABANDONED, COMPLETED */ {
+    active: boolean;
+    label: string /* e.g. 'Abandoned', 'Completed */;
+  };
+}
+
+// (filterKey, filterState) => setTypesFilter({ ...typesFilter, [filterKey]: filterState })
+export type FinanceReportOnChangeCallback = (filterKey: string, filterState: FilterState) => void;
+
+// setDirty: () => !isDirty && setIsDirty(true)
+export type FilterPageOnChangeCallback = () => void;
+
+// type ContainerOnChangeCallback = FinanceReportOnChangeCallback | FilterPageOnChangeCallback;
+export type ContainerOnChangeCallback =
+  | ((filterKey: string, filterState: FilterState) => void) // Finance Report
+  | (() => void); // PureFilterPage and TransactionFilter
```

Comment:
I would suggest defining this more generically as 

`export type ContainerOnChangeCallback = (filterKey?: string, filterState?: FilterState) => void`

There needs to be some uniformity to the callback's shape since the `FilterGroup` component will always be calling it with the same parameters, the only difference is whether the container cares about them or not

Reply:
I think my main goal -- because it was the hardest thing for me to understand reading these files -- was trying to convey that in some cases `onChange` is just calling `setDirty` and that passed argument -- which particularly reading the `FilterGroup` defined `onChange` would make me assume it's always doing something significant -- is completely dropped; completely irrelevant.

But maybe my original types didn't do that very clearly either... or maybe it's not even something that can/should be expressed in Types since it's not exactly about the function signature? Or the optional parameters makes it clear enough? 🤔 I'm not really sure...!

Reply:
I see your point! But we could probably throw in a comment instead to clarify that